components/brave_shields/content/browser/ad_block_custom_filters_provider.cc

```diff
@@ -5,14 +5,42 @@
 
 #include "brave/components/brave_shields/content/browser/ad_block_custom_filters_provider.h"
 
+#include <optional>
+#include <string>
 #include <utility>
 #include <vector>
 
+#include "base/strings/strcat.h"
+#include "base/strings/string_tokenizer.h"
 #include "base/task/single_thread_task_runner.h"
 #include "brave/components/brave_shields/core/browser/ad_block_filters_provider_manager.h"
 #include "brave/components/brave_shields/core/common/pref_names.h"
 #include "components/prefs/pref_service.h"
 
+namespace {
+
+std::optional<std::string> RemoveFiltersForHost(
+    const std::string& host,
+    const std::string& custom_filters) {
+  if (host.empty()) {
+    return std::nullopt;
+  }
+
+  std::string result{custom_filters};
+  const auto starts_with_str = base::StrCat({host, "##"});
```

Comment:
@ShivanKaul, I think we need to check the rule's content as well, I mean if there is a custom scriplets rule that looks like `host##+js(...)` we need to preserve it. I beleive the `content picker` can produce only specific content of rules and we can filter the custom rules more leniently.

Reply:
Good point, but let's create a separate issue for that. I'm worried that it will get tricky if we want to check for all cases here. The custom scriptlet case though can be easily checked, you're right. Let's address in follow-up.